### PR TITLE
use rbac.authorization.k8s.io/v1 for ClusterRole/Bindings where k8s >…

### PR DESCRIPTION
**Pull request by AdallomRoy**
…= 1.17

### CODE REVIEW

**Comment by AdallomRoy** [comment]
Fixed #516
Based on https://github.com/external-secrets/kubernetes-external-secrets/pull/581, but fixed tests.

File: charts/kubernetes-external-secrets/templates/rbac.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {{- if .Values.rbac.create -}}
+  {{- if semverCompare ">=1.17-0" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (moolen) [inline charts/kubernetes-external-secrets/templates/rbac.yaml:2]:
```suggestion
  {{- if semverCompare ">=1.17.0" .Capabilities.KubeVersion.GitVersion -}}
```

File: charts/kubernetes-external-secrets/templates/rbac.yaml [hunk 2]
```diff
@@ -31,7 +35,11 @@ rules:
     verbs: ["create"]
   {{- end }}
 ---
+{{ if semverCompare ">=1.17-0" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (moolen) [inline charts/kubernetes-external-secrets/templates/rbac.yaml:38]:
```suggestion
{{ if semverCompare ">=1.17.0" .Capabilities.KubeVersion.GitVersion -}}
```

File: charts/kubernetes-external-secrets/templates/rbac.yaml [hunk 3]
```diff
@@ -49,7 +57,11 @@ subjects:
     namespace: {{ .Release.Namespace | quote }}
     kind: ServiceAccount
 ---
+{{ if semverCompare ">=1.17-0" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (moolen) [inline charts/kubernetes-external-secrets/templates/rbac.yaml:60]:
```suggestion
{{ if semverCompare ">=1.17.0" .Capabilities.KubeVersion.GitVersion -}}
```

**Review by moolen — CHANGES_REQUESTED** [review]
Hey, thanks for picking this up!

I think there's a typo in the [semverCompare](https://helm.sh/docs/chart_template_guide/function_list/#semvercompare) function. I tried it out locally, with `>=1.17.0` it works fine!

For the record: `rbac.authorization.k8s.io/v1beta1` was deprecated with 1.17 and will be removed with 1.22

**Comment by AdallomRoy** [comment]
Thanks for your CR. Fixed.

**Review by moolen — APPROVED** [review]
thank you for your contribution :heart: !

**Pull request merged**